tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Driver;
+
+use Doctrine\DBAL\Driver\Statement;
+use Doctrine\DBAL\Driver\StatementIterator;
+
+class StatementIteratorTest extends \Doctrine\Tests\DbalTestCase
+{
+    public function testGettingIteratorDoesNotCallFetch()
+    {
+        $stmt = $this->createMock(Statement::class);
+        $stmt->expects($this->never())->method('fetch');
+
+        $stmtIterator = new StatementIterator($stmt);
+        $stmtIterator->getIterator();
+    }
+
+    public function testIterationCallsFetchOncePerStep()
+    {
+        $values = ['foo', '', 'bar', '0', 'baz', 0, 'qux', null, 'quz', false];
```

Comment:
We need to add one more value after `false` to prove that fetching `false` terminates iteration, while other empty values don't (see original example).